File: NotificationSystem/NotificationSystem.Web/Controls_Find/ctrAdmin_Find.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using System.Data;
using NotificationSystem.NotificationSystem.Data.NotificationSystemTableAdapters;
using static NotificationSystem.NotificationSystem.Data.NotificationSystem;
using NotificationSystem.NotificationSystem.Data.Classes;

namespace NotificationSystem.NotificationSystem.Web.Controls_Find
{
	public partial class ctrAdmin_Find : System.Web.UI.UserControl
	{
        private string m_UserName = string.Empty;

        public string UserName
        {
            get
            {
                return m_UserName;
            }
            set
            {
                m_UserName = value;
            }
        }

        protected void Page_Load(object sender, EventArgs e)
		{
            clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
            AdminDataTable tblAdmin = new AdminDataTable();

            try
            {
                if (Request.Form["ctl00$MainContent$ctrSearch_Admin_Find$btnSearch"] == "Search")
                    this.lblUserName.Text = "ID" + m_UserName;

                if ((Page.IsPostBack) & this.lblUserName.Text.Length > 0)
                    theNotificationSystem.GetAdmin(this.lblUserName.Text.Replace("Name", ""));
                else
                    tblAdmin = (AdminDataTable)theNotificationSystem.GetAdmin(UserName);


                this.lblSearchResult.Text = tblAdmin.Rows.Count.ToString();
                this.grdAdmin.DataSource = tblAdmin.DefaultView;
                this.grdAdmin.DataBind();
            }

            catch (Exception ex)
            {
                clsNotificationSystem_Web SendError = new clsNotificationSystem_Web();
                string NotificationBody = ex.Message + "  " + ex.StackTrace;
                SendError.SendMailMessage(NotificationBody);
                Response.Redirect("ErrorPage.aspx", false);
            }

        }
        private void grdAdmin_PageIndexChanging(object sender, GridViewPageEventArgs e)
        {
            try
            {
                if (!(Information.IsNothing(ViewState["columnname"]) | Information.IsNothing(ViewState["direction"])))
                {
                    DataView m_DataView = (DataView)this.grdAdmin.DataSource;

                    if (m_DataView == null)
                    {
                        m_DataView.Sort = ViewState["columnname"].ToString() + " " + ViewState["direction"].ToString();
                        this.grdAdmin.DataSource = m_DataView;
                    }
                }

                this.grdAdmin.PageIndex = e.NewPageIndex;
                this.grdAdmin.DataBind();
            }
            catch (Exception ex)
            {
                throw;
            }
        }

        private void grdAdmin_Sorting(object sender, GridViewSortEventArgs e)
        {
            try
            {
                DataView m_Dataview = (DataView)grdAdmin.DataSource;

                if (m_Dataview == null)
                {
                    m_Dataview.Sort = e.SortExpression + " " + ConvertSortDirection(e);
                    this.grdAdmin.DataSource = m_Dataview;
                    this.grdAdmin.DataBind();
                }
            }
            catch (Exception ex)
            {
                throw;
            }
        }

        private string ConvertSortDirection(System.Web.UI.WebControls.GridViewSortEventArgs e)
        {
            ViewState.Add("columnname", e.SortExpression);

            if ((ViewState["direction"] == null))
                ViewState.Add("direction", "asc");
            else
                ViewState["direction"] = Interaction.IIf(ViewState["direction"].ToString().ToLower() == "desc", "asc", "desc");

            return ViewState["direction"].ToString();
        }
        public void CreateExcelFiles()
        {
            try
            {
                Response.Clear();
                Response.Buffer = true;
                Response.AddHeader("content-disposition", "attachment;filename=Admin_Information.xls");
                Response.Charset = "";
                Response.ContentType = "application/vnd.ms-excel";
                StringWriter sw = new StringWriter();
                HtmlTextWriter hw = new HtmlTextWriter(sw);
                grdAdmin.AllowPaging = false;
                this.grdAdmin.DataBind();

                for (int y = 0; y <= 12; y++)
                    grdAdmin.HeaderRow.Cells[y].Style.Add("background-color", "#cfdbe6");

                for (int x = 0; x <= grdAdmin.Rows.Count - 1; x++)
                {
                    for (int y = 0; y <= 12; y++)
                    {
                        if (x % 2 > 0)
                            grdAdmin.Rows[x].Cells[y].Style.Add("background-color", "#e0ebea");
                    }
                }

                grdAdmin.Attributes.Add("style", "vnd.ms-excel.numberformat:@");
                grdAdmin.RenderControl(hw);
                Response.Output.Write(sw.ToString());
                Response.Flush();
                Response.End();
            }




            catch (Exception ex)
            {
                throw;
            }
        }

        private void btnSubmit_Click(object sender, EventArgs e)
        {
            CreateExcelFiles();
        }

    }
}

File: NotificationSystem/NotificationSystem.Web/Controls_Find/ctrAdmin_Find.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.IO
Imports Microsoft.VisualBasic
Imports System.Data
Imports NotificationSystem.NotificationSystem.Data.Classes
Imports NotificationSystem.NotificationSystem.Data.NotificationSystem

Namespace NotificationSystem.NotificationSystem.Web.Controls_Find
    Partial Public Class ctrAdmin_Find
        Inherits UserControl
        Private m_UserName As String = String.Empty

        Public Property UserName As String
            Get
                Return m_UserName
            End Get
            Set(value As String)
                m_UserName = value
            End Set
        End Property

        Protected Sub Page_Load(ByVal sender As Object, e As System.EventArgs) Handles Me.Load

            If Page.IsPostBack Then
                Try
                    If m_UserName.Length > 0 Then
                        lblUserName.Text = "UserName" & m_UserName
                        If m_UserName.Length = 2 Then Exit Sub

                        If New clsNotificationSystem().GetAdmin(m_UserName).Rows.Count = 0 Then Exit Sub

                        If Request.Form("ctl00$MainContent$ctrSearch_Admin_Find$btnSearch") = "Search" Then lblUserName.Text = "UserName" & m_UserName

                    End If


                    Dim theNotificationSystem As New clsNotificationSystem
                    Dim tblAdmin As AdminDataTable = New AdminDataTable

                    If Page.IsPostBack AndAlso lblSearchResult.Text.Length > 0 Then

                        tblAdmin = CType(theNotificationSystem.GetAdmin(m_UserName), AdminDataTable)
                    Else
                        tblAdmin = CType(theNotificationSystem.GetAdmins(), AdminDataTable)
                    End If


                    lblSearchResult.Text = tblAdmin.Rows.Count.ToString()
                    grdAdmin.DataSource = tblAdmin.DefaultView
                    grdAdmin.DataBind()

                Catch ex As Exception
                    Dim SendError As clsNotificationSystem_Web = New clsNotificationSystem_Web()
                    Dim NotificationBody = ex.Message & "  " & ex.StackTrace
                    SendError.SendMailMessage(NotificationBody)
                    Response.Redirect("ErrorPage.aspx", False)
                End Try
            End If
        End Sub



        Private Function ConvertSortDirection(e As GridViewSortEventArgs) As String
            MyBase.ViewState.Add("columnname", e.SortExpression)

            If MyBase.ViewState("direction") Is Nothing Then
                MyBase.ViewState.Add("direction", "asc")
            Else
                MyBase.ViewState("direction") = IIf(Equals(MyBase.ViewState("direction").ToString().ToLower(), "desc"), "asc", "desc")
            End If

            Return MyBase.ViewState("direction").ToString()
        End Function
        Public Sub CreateExcelFiles()
            Try
                Response.Clear()
                Response.Buffer = True
                Response.AddHeader("content-disposition", "attachment;filename=Admin_Information.xls")
                Response.Charset = ""
                Response.ContentType = "application/vnd.ms-excel"
                Dim sw As StringWriter = New StringWriter()
                Dim hw As HtmlTextWriter = New HtmlTextWriter(sw)
                grdAdmin.AllowPaging = False
                grdAdmin.DataBind()

                For y = 0 To 12
                    grdAdmin.HeaderRow.Cells(y).Style.Add("background-color", "#cfdbe6")
                Next

                For x As Integer = 0 To grdAdmin.Rows.Count - 1
                    For y = 0 To 12
                        If x Mod 2 > 0 Then grdAdmin.Rows(x).Cells(y).Style.Add("background-color", "#e0ebea")
                    Next
                Next

                grdAdmin.Attributes.Add("style", "vnd.ms-excel.numberformat:@")
                grdAdmin.RenderControl(hw)
                Response.Output.Write(sw.ToString())
                Response.Flush()
                Response.End()




            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub

        Private Sub btnSubmit_Click(sender As Object, e As EventArgs)
            CreateExcelFiles()
        End Sub

        Protected Sub grdAdmin_PageIndexChanging(sender As Object, e As GridViewPageEventArgs)
            Try
                If Not (IsNothing(MyBase.ViewState("columnname")) Or IsNothing(MyBase.ViewState("direction"))) Then
                    Dim m_DataView = CType(grdAdmin.DataSource, DataView)

                    If m_DataView Is Nothing Then
                        m_DataView.Sort = MyBase.ViewState("columnname").ToString() & " " & MyBase.ViewState("direction").ToString()
                        grdAdmin.DataSource = m_DataView
                    End If
                End If

                grdAdmin.PageIndex = e.NewPageIndex
                grdAdmin.DataBind()
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub

        Protected Sub grdAdmin_Sorting(sender As Object, e As GridViewSortEventArgs)

            Try
                Dim m_Dataview = CType(grdAdmin.DataSource, DataView)

                If m_Dataview Is Nothing Then
                    m_Dataview.Sort = e.SortExpression & " " & ConvertSortDirection(e)
                    grdAdmin.DataSource = m_Dataview
                    grdAdmin.DataBind()
                End If
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub
    End Class
End Namespace
